int32 x_coord
int32 y_coord
int32 radius
int8 id
